features/graph/api/src/main/java/org/opennms/alec/features/graph/api/GraphProvider.java

```diff
@@ -37,4 +37,13 @@
 
     void withReadOnlyGraph(Consumer<OceGraph> consumer);
 
+    void registerGraphChangeListener(GraphChangedListener listener);
+
+    default OceGraph getOceGraph() {
```

Comment:
The graph is not thread-safe, so it should only be read on the context of a `withReadOnlyGraph` call.

Reply:
fixed

Reply:
fixed